File: app/src/main/java/com/codingbuffalo/aerialdream/data/Apple2019Repository.java
package com.codingbuffalo.aerialdream.data;

import android.content.Context;

import com.codingbuffalo.aerialdream.R;

import java.util.List;

public class Apple2019Repository extends VideoRepository {
    @Override
    public List<Apple2017Video> fetchVideos(Context context) {
        Wrapper wrapper = parseJson(context, R.raw.tvos13, Wrapper.class);
        return wrapper.assets;
    }

    private static class Wrapper {
        private List<Apple2017Video> assets;
    }
}

File: app/src/main/java/com/codingbuffalo/aerialdream/data/Apple2019Repository.kt
package com.codingbuffalo.aerialdream.data

import android.content.Context
import com.codingbuffalo.aerialdream.R

class Apple2019Repository : VideoRepository() {
    override fun fetchVideos(context: Context): List<Video> {
        val wrapper = parseJson(context, R.raw.tvos13, Wrapper::class.java)
        return wrapper.assets!!
    }

    private class Wrapper {
        val assets: List<Apple2019Video>? = null
    }
}